jgrapht-core/src/test/java/org/jgrapht/alg/shortestpath/YenShortestPathIteratorTest.java

```diff
@@ -0,0 +1,345 @@
+
+        assertTrue(it.hasNext());
+        performAssertion(it, 4.0, true);
+        performAssertion(it, 4.0, false);
+    }
+
+    @Test
+    public void testCyclicGraph3() {
+        Graph<Integer, DefaultWeightedEdge> graph = new SimpleDirectedWeightedGraph<>(DefaultWeightedEdge.class);
+        readGraph(graph, cyclicGraph3);
+        Integer source = 1;
+        Integer target = 3;
+        YenShortestPathIterator<Integer, DefaultWeightedEdge> it =
+                new YenShortestPathIterator<>(graph, source, target);
+
+        assertTrue(it.hasNext());
+        performAssertion(it, 2.0, false);
+    }
+
+
+    @Test
+    public void testPseudoGraph() {
+        Graph<Integer, DefaultWeightedEdge> graph = new DirectedWeightedPseudograph<>(DefaultWeightedEdge.class);
+        readGraph(graph, multigraph);
+        Integer source = 1;
+        Integer target = 5;
+        YenShortestPathIterator<Integer, DefaultWeightedEdge> it =
+                new YenShortestPathIterator<>(graph, source, target);
+
+        assertTrue(it.hasNext());
+        performAssertion(it, 4.0, true);
+        performAssertion(it, 7.0, true);
+        performAssertion(it, 10.0, false);
+    }
+
+    @Test
+    public void testNotShortestPathEdgesGraph() {
+        Graph<Integer, DefaultWeightedEdge> graph = new SimpleDirectedWeightedGraph<>(DefaultWeightedEdge.class);
+        readGraph(graph, notShortestPathEdgesGraph);
+        Integer source = 1;
+        Integer target = 2;
+        YenShortestPathIterator<Integer, DefaultWeightedEdge> it
+                = new YenShortestPathIterator<>(graph, source, target);
+
+        assertTrue(it.hasNext());
+        performAssertion(it, 1.0, false);
+    }
+
+    @Test
+    public void testRandomGraphs() {
+        int n = 100;
+        double p = 0.5;
+        for (int i = 0; i < 1000; i++) {
+            SimpleDirectedWeightedGraph<Integer, DefaultWeightedEdge> graph
+                    = new SimpleDirectedWeightedGraph<>(DefaultWeightedEdge.class);
+            graph.setVertexSupplier(SupplierUtil.createIntegerSupplier());
+            getRandomGraph(graph, n, p);
+            Integer source = (int) (Math.random() * n);
+            Integer target = (int) (Math.random() * n);
+            assertCorrectness(graph, source, target);
+        }
+    }
+
+    /**
+     * If the overall number of paths between {@code source}
+     * and {@code target} is denoted by $n$ and the value of
+     * {@code #NUMBER_OF_PATH_TO_ITERATE} is denoted by $m$
+     * then the method iterates over $p = min\{n, m\}$ such paths
+     * and verifies that they are built correctly. Additionally
+     * method checks that are returned in the increasing order by
+     * weight.
+     *
+     * @param graph  graph the iterator is being tested on
+     * @param source source vertex
+     * @param target target vertex
+     */
+    private void assertCorrectness(Graph<Integer, DefaultWeightedEdge> graph,
```

Comment:
It's better to rename this function to `testOnRandomGraph` to reflect the fact that we test the algorithm on the random graph instance.